.github/workflows/check-dependencies.yaml

```diff
@@ -0,0 +1,110 @@
+name: "Validate wolfssl-sys dependency version"
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    outputs:
+      dependency_status:
+        description: "Is wolfssl's versioned dependency on wolfssl-sys up to date [released, pending, unreleased, out-of-date]"
+        value: ${{ jobs.query.outputs.dependency_status }}
+      dependency_version:
+        description: "Current value of wolfssl's versioned dependency on wolfssl-sys"
+        value: ${{ jobs.query.outputs.dependency_version }}
+      wolfssl_sys_release_status:
+        description: "Current wolfssl-sys status: [released, pending, unreleased]"
+        value: ${{ jobs.query.outputs.wolfssl_sys_release_status }} 
+      wolfssl_sys_release_version:
+        description: "Current wolfssl-sys wolfssl release"
+        value: ${{ jobs.query.outputs.wolfssl_sys_release_version }}
+
+jobs:
+  query:
+    runs-on: ubuntu-latest
+    outputs:
+      dependency_status: ${{ steps.validate_dependency.outputs.status }}
+      dependency_version: ${{ steps.get-dep.outputs.version }}
+      wolfssl_sys_release_status: ${{ steps.validate_sys_release.outputs.status }}
+      wolfssl_sys_release_version: ${{ steps.get-sys-release.outputs.version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get wolfssl-sys current release version
+        id: get-sys-release
+        shell: bash
+        run: |
+          VER=$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "wolfssl-sys") | .version')
+          echo version=${VER} >> $GITHUB_OUTPUT
+      - name: Get wolfssl-sys current release treesha
+        id: get-sys-release-treesha
+        shell: bash
+        run: |
+          TAG="wolfssl-sys-${{ steps.get-sys-release.outputs.version }}"
+          if ! git fetch origin --depth=1 tag "$TAG" ; then
+              echo "tag $TAG not found, assuming pending release"
+              echo sha=pending >> $GITHUB_OUTPUT
+              exit 0
+          fi
+
+          TREESHA=$(git ls-tree --object-only $TAG wolfssl-sys)
+          echo TREESHA=$TREESHA
```

Comment:
FYI, we could use github step/job summary to print these things.
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary

Might be easier to take a look for these release workflows

Reply:
Ooh, nice, let me have a play with that!

Reply:
They appear in an overall job summary, like https://github.com/expressvpn/wolfssl-rs/actions/runs/9594333419.

Which is nice but I think we might want the per-step echoes when debugging too?

Reply:
https://github.com/expressvpn/wolfssl-rs/actions/runs/9594476612 is what I ended up with, basically the output of the two `validate` steps goes there but the various probes log with echo.